streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -560,6 +561,52 @@ object ZTransducer {
   def last[O]: ZTransducer[Any, Nothing, O, Option[O]] =
     foldLeft[O, Option[O]](Option.empty[O])((_, a) => Some(a))
 
+  def magicSequence[R, E, I, O](n: Int)(f: Chunk[I] => ZTransducer[R, E, I, O]): ZTransducer[R, E, I, O] =
```

Comment:
I think this can be more general in that we could have a combinator that "decides" on a transducer after the first element has been emitted and is reset on a push(None).
Should be add something like this? Signature would be:

```
ZTransducer[R, E, I, O] {
  def replace[R1 <: R, E1 >: E, I1 <: I, O1](f: O => ZTransducer[R1, E1, I1, O1]): ZTransducer[R1, E1, I1, O1]
}
```

Reply:
This is kind of similar to `ZStream#peel` but in a transducer form. The infix form would be `self >>> ZStream.magicSequence` no?

Either way would be great if we can just find a better name before merging. Something around branching, headers?

Reply:
True, so we probably don't need an additional combinator.

What about `branchAfter` for the name? As in "after reading n values, branch"